File: challenge-231/steven-wilson/perl/ch-01.pl
#!/usr/bin/env perl

use 5.12.0;
use Test2::V0 -no_srand;
use List::Util qw/ min max /;

is( pluck_min_max( ( 3, 2, 1, 4 ) ),
    array {
        item 3;
        item 2;
    },
    "Example 1"
);
is( pluck_min_max( ( 3, 1 ) ), -1, "Example 2" );
is( pluck_min_max( ( 2, 1, 3 ) ),
    array {
        item 2;
    },
    "Example 3"
);
done_testing();

sub pluck_min_max {
    my @elements = @_;
    return -1 if ( scalar @elements < 3 );
    my $min      = min @elements;
    my $max      = max @elements;
    my @result   = grep { !( ( $min == $_ ) || ( $max == $_ ) ) } @elements;
    return \@result;
}

File: challenge-231/steven-wilson/javascript/ch-01.js
"use strict";


function pluckMinMax(elements) {
    if (elements.length < 3) {
        return -1;
    }
    let removePositions = [];
    removePositions.push(elements.indexOf(Math.min(...elements)));
    removePositions.push(elements.indexOf(Math.max(...elements)));
    for (let position of removePositions.sort((a, b) => b - a)) {
        elements.splice(position, 1);
    }
    return elements;
}
